### ISSUE TITLE
media type application/problem+json in additional responses

### ISSUE THREAD

**Issue by neveritt** [tiangolo/fastapi#3122]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI
from fastapi.responses import JSONResponse
from pydantic import BaseModel

app = FastAPI()

class JsonApiResponse(JSONResponse):
    media_type = "application/vnd.api+json"

class JsonApiError(BaseModel):
    type: str
    title: str
    detail: str

@app.get(
    "/a/{id}",
    response_class=JsonApiResponse,
    responses={422: {
            "model": JsonApiError,
            "media_type": "application/problem+json"
        },
    },
)
async def a(id):
    pass  # pragma: no cover
```

### Description

I would like to use a Pydantic model to specify an error message as conforming to
[application/problem+json](https://tools.ietf.org/html/rfc7807), i.e. setting the media type to application/problem+json.

However, afaik, in the openAPI specification the media type is fixed to "application/json", unless "response_class" is set with `media_type` as in the above example. However, this results in all responses having that media type.

### The solution you would like

I would like to be able to set the media type individually for each response. 

### Describe alternatives you've considered

I propose that in the response `dict`'s the key "media_type" should be used to set the media type.

Naturally, if "media_type" is present it should take precedence over response_class which in turn overrides the default "application/json".

```Python
@app.get(
    "/a/{id}",
    response_class=JsonApiResponse,
    responses={422: {
            "model": JsonApiError,
            "media_type": "application/problem+json"
        },
    },
)
async def a(id):
    pass  # pragma: no cover
```

I have not come up with any other ideas on how to solve this.

In case you find my proposal reasonable, I'd be happy to provide a pull request.

### Environment

* OS: macOS
* FastAPI Version: 0.63.0
* Python version: 3.8.7

**Comment by Kludex** [comment]
Does this satisfy you?
```python
from fastapi import FastAPI
from fastapi.responses import JSONResponse
from pydantic import BaseModel

app = FastAPI()

class JsonApiResponse(JSONResponse):
    media_type = "application/vnd.api+json"

class JsonApiError(BaseModel):
    type: str
    title: str
    detail: str

@app.get(
    "/a/{id}",
    response_class=JsonApiResponse,
    responses={
        422: {
            "model": JsonApiError,
            "content": {"application/problem+json": {}}
        }
    }
)
async def a(id):
    pass  # pragma: no cover
```

Reference: https://fastapi.tiangolo.com/advanced/additional-responses/?h=responses#combining-information

**Comment by neveritt** [comment]
Such a solution generates the following for me:

```
"422": {
    "description": "Unprocessable Entity",
    "content": {
         "application/problem+json": {},
         "application/vnd.api+json": {
          "schema": {
              "$ref": "#/components/schemas/JsonApiError"
           }
        }
    }
}
```

While I am hoping to generate
```
"422": {
    "description": "Unprocessable Entity",
    "content": {
         "application/problem+json": {
          "schema": {
              "$ref": "#/components/schemas/JsonApiError"
           }
        }
    }
}
```

I'm pretty sure I could solve my problem by customizing the api as described in the [doc](https://fastapi.tiangolo.com/advanced/extending-openapi/?h=customize+open) though. However I was hoping for a more elegant solution

**Comment by neveritt** [comment]
Perhaps being able to override the _class_ of the response is a better proposal

```python
from fastapi import FastAPI
from fastapi.responses import JSONResponse
from pydantic import BaseModel

app = FastAPI()

class JsonApiResponse(JSONResponse):
    media_type = "application/vnd.api+json"

class JsonApiErrorResponse(JSONResponse):
    media_type = "application/problem+json"

class JsonApiError(BaseModel):
    type: str
    title: str
    detail: str

@app.get(
    "/a/{id}",
    response_class=JsonApiResponse,
    responses={422: {
            "model": JsonApiError,
            "class": JsonApiErrorResponse
        },
    },
)
async def a(id):
    pass  # pragma: no cover
```

**Comment by tiangolo** [comment]
Yep, currently, the way to customize it as you want would be manually modifying the generated OpenAPI.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
